app/views/pet_applications/show.html.erb

```diff
@@ -0,0 +1,37 @@
+<h1>Application Number: <%= @pet_application.id %></h1>
+<p><strong>Name:</strong> <%= @pet_application.name %></p>
+<p><strong>Address:</strong> <%= @pet_application.street %>, <%= @pet_application.city %>, <%= @pet_application.state %>, <%= @pet_application.zip %></p>
+<p><strong>Description:</strong> <%= @pet_application.description %></p>
+<p><strong>Status:</strong> <%= @pet_application.status %></p>
+
+<p><strong>Pets Applying For:</strong>
+<ul>
+  <% @pet_application.pets.each do |pet| %>
+    <li><%= link_to pet.name, "/pets/#{pet.id}" %></li>
+  <% end %>
+</ul><br>
+
+<% if @pet_application.status == "In Progress" %>
+  <p><strong>Add a Pet to this Application:</strong></p>
+  <%= form_with url: "/pet_applications/#{@pet_application.id}", method: :get, data: { turbo: false } do |f| %>
+    <%= f.label "Pet Name" %>
+    <%= f.text_field :pet_search %>
+    <%= f.submit "Search" %><br>
+  <% end %><br>
+
+  <% if @pets != nil %>
+    <% @pets.each do |pet| %>
+      <%= pet.name %> 
+      <%= button_to "Adopt this Pet", "/pet_applications/#{@pet_application.id}/pets/#{pet.id}", method: :post %>
```

Comment:
I just want to circle back to this button because I think this is the one we worked on together, but I could also be wrong. For this button, in order to make the route more restful, here is what it would look like.

```ruby
<%= button_to "Adopt this Pet", "/pet_applications/#{@pet_application.id}/pets", params: {pet_id: pet.id}, method: :post %>
```

What you'll notice is that we added a new key/value pair of `params: {pet_id: pet.id}`. This is the exact same `params` hash that we get access to in the controller, we are accessing it and adding to it directly when this button gets selected.

As backend developers, our goal is to get the data we need in order to do the things we need to do on the server, this is yet another way to populate the `params` hash.

The route in `routes.rb` would then be `"/pet_applications/:app_id/pets"`.

In the controller, our `params` hash would contain `{app_id: <some num>, pet_id: <some num>}` where `app_id` comes from the dynamic segment in the path, and `pet_id` comes from us adding it directly in the `button_to`.

Reach out if this doesn't make sense.